benchmark/ubench.c

```diff
@@ -438,6 +439,47 @@ static void do_ipc_get_put_benchmark(alloc_t *allocs, size_t num_allocs,
     }
 }
 
+static void do_umf_mem_props_benchmark(ze_context_handle_t context,
+                                       bool use_umf, alloc_t *allocs,
+                                       size_t num_allocs, size_t repeats) {
+    assert(context != NULL);
+
+    for (size_t r = 0; r < repeats * 10; ++r) {
+        for (size_t i = 0; i < num_allocs; ++i) {
+            if (use_umf) {
+                umf_memory_properties_handle_t props_handle = NULL;
+                umf_result_t res =
+                    umfGetMemoryPropertiesHandle(allocs[i].ptr, &props_handle);
+                (void)res;
+                assert(res == UMF_RESULT_SUCCESS);
+
+                umf_usm_memory_type_t type = UMF_MEMORY_TYPE_UNKNOWN;
+                res = umfGetMemoryProperty(
+                    props_handle, UMF_MEMORY_PROPERTY_POINTER_TYPE, &type);
+                assert(res == UMF_RESULT_SUCCESS);
+                if (type != UMF_MEMORY_TYPE_DEVICE) {
+                    fprintf(stderr,
+                            "error: unexpected alloc_props.type value: %d\n",
+                            type);
+                    exit(-1);
```

Comment:
Please in final version add benchmark to new framework.

Reply:
I removed the old version of the benchmark, but the new one should be done in a separate PR